orchagent/port/porthlpr.cpp

```diff
@@ -471,6 +457,11 @@ bool PortHelper::parsePortFec(PortConfig &port, const std::string &field, const
     port.fec.value = cit->second;
     port.fec.is_set = true;
 
+    if (value == PORT_FEC_AUTO)
```

Comment:
@dgsudharsan more specific config values should to be processed first and return immediately leaving the rest of the logic behind